ludwig/encoders/image_encoders.py

```diff
@@ -315,105 +244,783 @@ def output_shape(self) -> torch.Size:
+                    f"Instantiating torchvision image encoder: '{self.torchvision_model_type}' "
+                    "with no pretrained weights."
+                )
 
-        self._output_shape = (self.transformer.config.hidden_size,)
-        self.output_attentions = output_attentions
+        # get torchvision transforms object
+        transforms_obj = torchvision_model_registry[model_id].weights_class.DEFAULT.transforms()
+        self.num_channels = len(transforms_obj.mean)
+        self.crop_size = transforms_obj.crop_size
 
-    def forward(self, inputs: torch.Tensor, head_mask: torch.Tensor = None) -> Dict[str, torch.Tensor]:
-        output = self.transformer(inputs, head_mask=head_mask, output_attentions=self.output_attentions)
-        return_dict = {"encoder_output": output.pooler_output}
-        if self.output_attentions:
-            return_dict["attentions"] = output.attentions
-        return return_dict
+        logger.debug(f"  {model_id}")
+        # create pretrained model with pretrained weights or None for untrained model
+        self.model = self.create_model(weights=weights_specification, **kwargs)
 
-    @staticmethod
-    def get_schema_cls():
-        return ViTEncoderConfig
+        # remove final classification layer
+        self._remove_last_layer()
 
-    @property
-    def input_shape(self) -> torch.Size:
-        return torch.Size(self._input_shape)
+        # freeze parameters if requested
+        for p in self.model.parameters():
+            p.requires_grad_(trainable)
+
+    def forward(self, inputs: torch.Tensor) -> Dict[str, torch.Tensor]:
+        return {"encoder_output": self.model(inputs)}
+
+    @abstractmethod
+    def _remove_last_layer(self):
+        raise NotImplementedError()
 
     @property
     def output_shape(self) -> torch.Size:
-        return torch.Size(self._output_shape)
+        # create synthetic image and run through forward method
+        inputs = torch.randn([1, *self.input_shape])
+        output = self.model(inputs)
+        return torch.Size(output.shape[1:])
+
+    @property
+    def input_shape(self) -> torch.Size:
+        # expected shape after all pre-processing
+        # len(transforms_obj.mean) determines the number of channels
+        # transforms_obj.crop_size determines the height and width of image
+        # [num_channels, height, width]
+        return torch.Size([self.num_channels, *(2 * self.crop_size)])
+
+
+# TVModelVariant(variant_id, create_model_function, model_weights)
+#   variant_id: model variant identifier
+#   create_model_function: TorchVision function to create model class
+#   model_weights: Torchvision class for model weights
+
+ALEXNET_VARIANTS = [
+    TVModelVariant("base", tvm.alexnet, tvm.AlexNet_Weights),
+]
+
+
+@register_torchvision_variant(ALEXNET_VARIANTS)
+@register_encoder("alexnet_torch", IMAGE)
+class TVAlexNetEncoder(TVBaseEncoder):
+    # specify base torchvision model
+    torchvision_model_type: str = "alexnet_torch"
```

Comment:
nit: Let's make this an official TVBaseEncoder abstract method that subclasses are required to implement, i.e.

```python
# In the base class
def get_torchvision_model_type():
    return NotImplementedError()

# In the subclass
def get_torchvision_model_type():
    return "alexnet_torch"

# Callers:
encoder.get_torchvision_model_type()
```

Reply:
@justinxzhao  next week, I'd like to discuss this.  As noted in another comment, I think this construct may not work.

Reply:
Discussed with Justin.  No action at this time.